### PR TITLE
feat(forms): improve types for `AbstractControl#status`

### PR DESCRIPTION
**Pull request by Toxicable**
## PR Checklist
- [x] The commit message follows our guidelines: 
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
```

## What is the current behavior?
`AbstractControl#status` is currently typed to string where as it has four distinct cases of it's value

## What is the new behavior?
`AbstractControl#status` now is of type `'VALID' | 'INVALID' | 'PENDING' | 'DISABLED'` giving slightly better support for the developer

## Does this PR introduce a breaking change?
```
[x] Yes (maybe)
[ ] No
```
If someone is comparing `AbstractControl#status` to a string other than the ones it will be typed to then after this change TS will throw an error at compile time

Duplicate of #16841 - since I couldn't fix circle ci validating old commits messages

### CODE REVIEW

File: packages/forms/src/model.ts [hunk 1]
```diff
@@ -229,7 +229,9 @@ export abstract class AbstractControl {
    * Emits an event every time the validation status of the control
    * is re-calculated.
    */
-  get statusChanges(): Observable<any> { return this._statusChanges; }
+  get statusChanges(): Observable<'VALID'|'INVALID'|'PENDING'|'DISABLED'> {
+    return <EventEmitter<any>>this._statusChanges;
```
Comment (Toxicable) [inline packages/forms/src/model.ts]: TS throws here if we don't do this silly cast.
It appears there is some issue where we can't assign `EventEmitter<T>` to `Observable<T>`

**Comment by snerks** [comment]
Consider refactoring to a type:

```typescript
export type AbstractControlStatus = 'VALID'|'INVALID'|'PENDING'|'DISABLED';
```

**Comment by Toxicable** [comment]
The issue with that is it'll be yet another export, weather it's public or not I'm unsure but either way cutting down on exports is always good

**Comment by snerks** [comment]
OK. So, keep the `type` but remove the `export` keyword?

**Comment by kyr0** [comment]
Merging this would help our team maintaining a clean coding style 👍

**Comment by AndrewKushnir** [comment]
Closing this PR in favor of more recent #38406 and #38906. Thank you.

**Pull request closed without merging**